packages/components/src/internal/components/editable/EditableGridPanel.tsx

```diff
@@ -59,29 +84,119 @@ export class EditableGridPanel extends ReactN.Component<Props, any, GlobalAppSta
     }
 
     getModel(): QueryGridModel {
-        const { model } = this.props;
+        const model = this.getActiveModel(this.props);
 
         // need to access this.global directly to connect this component to the re-render cycle
         return this.global.QueryGrid_models.get(model.getId());
     }
 
+    getModelsAsList(props: Props): List<QueryGridModel> {
+        const { model, models } = props;
+        if (models) return List.isList(models) ? List(models.toArray()) : List<QueryGridModel>([models]);
```

Comment:
```suggestion
        if (models) return List.isList(models) ? models : List<QueryGridModel>([models]);
```
If it's already a list we don't need to make a new one, do we?

Reply:
We don't, but unfortunately typescript service is not smart enough to read the if check and will error out with incompatible return type.

Reply:
Can you use `models as List` perhaps?  No big deal, of course, just inefficient.